### PR TITLE
[YSQL] #1126  Row-level partitioning support in YSQL

### PR DESCRIPTION
**Pull request by deeps1991**
Picking this up from https://github.com/yugabyte/yugabyte-db/pull/4761

### CODE REVIEW

File: src/postgres/src/test/regress/expected/yb_pg_partition_prune.out [hunk 1]
```diff
@@ -0,0 +1,2854 @@
+--
```
Comment (m-iancu) [inline src/postgres/src/test/regress/expected/yb_pg_partition_prune.out:1]: the .sql file for this does not seem to be here.

File: java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgRegressPartitions.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+@RunWith(value=YBTestRunnerNonTsanOnly.class)
+public class TestPgRegressPartitions extends BasePgSQLTest {
+  @Override
+  public int getTestMethodTimeoutSec() {
+    return 1800;
+  }
+
+  @Test
+  public void testPgRegressTypes() throws Exception {
```
[the remark this answers is not in the document]
Comment (deeps1991) [reply java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgRegressPartitions.java]: Done

File: src/postgres/src/test/regress/expected/yb_pg_partition_prune.out (hunk 1 above)
Comment (deeps1991) [reply src/postgres/src/test/regress/expected/yb_pg_partition_prune.out:1]: Added it in the next patch, really sorry about missing it.

**Comment by deeps1991** [comment]
#5179

**Pull request merged**